pytest 6: pylint still shows not-callable error for builtin pytest marks even after #7473 and #7476

I'm experiencing #7473 after the fix in #7476 with the following:

```python
import pytest

@pytest.mark.parametrize('a', [1, 2])
def test_a(a):
    pass
```

Where pylint still shows `not-callable` for `pytest.mark.parametrize`:

```
$ pylint -E test_pytest_7473.py
************* Module test_pytest_7473
test_pytest_7473.py:3:1: E1102: pytest.mark.parametrize is not callable (not-callable)
```

Environment:

- Python 3.8
- pytest==6.0.0
- pylint==2.5.3

FYI @bluetech

--- a/changelog/7558.bugfix.rst
+++ b/changelog/7558.bugfix.rst
@@ -1,0 +1,2 @@
+Fix pylint ``not-callable`` lint on ``pytest.mark.parametrize()`` and the other builtin marks:
+``skip``, ``skipif``, ``xfail``, ``usefixtures``, ``filterwarnings``.

--- a/src/_pytest/mark/structures.py
+++ b/src/_pytest/mark/structures.py
@@ -4,7 +4,6 @@ import typing
 import warnings
 from typing import Any
 from typing import Callable
-from typing import cast
 from typing import Iterable
 from typing import List
 from typing import Mapping
@@ -473,14 +472,13 @@ class MarkGenerator:
 
     # See TYPE_CHECKING above.
     if TYPE_CHECKING:
-        # Using casts instead of type comments intentionally - issue #7473.
         # TODO(py36): Change to builtin annotation syntax.
-        skip = cast(_SkipMarkDecorator, None)
-        skipif = cast(_SkipifMarkDecorator, None)
-        xfail = cast(_XfailMarkDecorator, None)
-        parametrize = cast(_ParametrizeMarkDecorator, None)
-        usefixtures = cast(_UsefixturesMarkDecorator, None)
-        filterwarnings = cast(_FilterwarningsMarkDecorator, None)
+        skip = _SkipMarkDecorator(Mark("skip", (), {}))
+        skipif = _SkipifMarkDecorator(Mark("skipif", (), {}))
+        xfail = _XfailMarkDecorator(Mark("xfail", (), {}))
+        parametrize = _ParametrizeMarkDecorator(Mark("parametrize ", (), {}))
+        usefixtures = _UsefixturesMarkDecorator(Mark("usefixtures ", (), {}))
+        filterwarnings = _FilterwarningsMarkDecorator(Mark("filterwarnings ", (), {}))
 
     def __getattr__(self, name: str) -> MarkDecorator:
         if name[0] == "_":

